src/chunk_list.cpp

```diff
@@ -721,32 +736,26 @@ void chunk_swap_lines(chunk_t *pc1, chunk_t *pc2)
 } // chunk_swap_lines
 
 
-/**
- * Gets the next non-vbrace chunk
- */
-chunk_t *chunk_get_next_nvb(chunk_t *cur, chunk_nav_t nav)
+chunk_t *chunk_get_next_nvb(chunk_t *cur, const chunk_nav_t nav)
 {
-   chunk_t *pc = cur;
+   return(chunk_get_nvb(cur, nav, true));       // \todo check this
+}
 
-   do
-   {
-      pc = chunk_get_next(pc, nav);
-   } while (chunk_is_vbrace(pc));
-   return(pc);
+
+chunk_t *chunk_get_prev_nvb(chunk_t *cur, const chunk_nav_t nav)
+{
+   return(chunk_get_nvb(cur, nav, false));
 }
 
 
-/**
- * Gets the prev non-vbrace chunk
- */
-chunk_t *chunk_get_prev_nvb(chunk_t *cur, chunk_nav_t nav)
+static chunk_t *chunk_get_nvb(chunk_t *cur, const chunk_nav_t nav, const bool dir)
 {
    chunk_t *pc = cur;
 
    do
    {
-      pc = chunk_get_prev(pc, nav);
-   } while (chunk_is_vbrace(pc));
+      pc = (dir == true) ? chunk_get_next(pc, nav) : chunk_get_prev(pc, nav);
```

Comment:
I don't think that the idea is bad, it is just not going the full circle.
* `chunk_get_prev_nvb` and `chunk_get_next_nvb` should be completely removed and replaced with the new `chunk_get_nvb` (or better `chunk_get_non_vbrace`, I still would have too look up what a vbrace is, but at least I will know is some kind of brace)
* the direction check could be a switch that has loop bodies for every case

> Optimizing for performance should only be done as last step if there are performance issues

I hear that all the time but there never actually is a planned last step with open source. Is either improved on or dead forever. 
Unless we know that there is some compiler magic happening that fixes this it will be slower (even if no one will notice it).